examples/kubernetes/recursive-custom-types/rg.yaml

```
@@ -0,0 +1,42 @@
+---
+apiVersion: kro.run/v1alpha1
+kind: ResourceGraphDefinition
+metadata:
+  name: company.kro.run
+spec:
+  schema:
+    types:
+      Person:
+        name: string
+        role: "Role"
+      Division:
+        name: string
+        employees: "[]Person"
+      Role:
+        name: string
+        seniority: string
```

Will `types` get inside the `CompanyConfig` CRD? or is it a construct that stays private within the RGD?

From birds view, it would be interesting to keep it as a construct and exclude it from the `CompanyConfig`  CRD. I would hence advocate for moving it under spec

```suggestion
  types:
    Person:
      name: string
      role: "Role"
    Division:
      name: string
      employees: "[]Person"
    Role:
      name: string
      seniority: string
  schema:
```